src/Symfony/Component/Notifier/Bridge/Firebase/FirebaseTransport.php

```diff
@@ -53,21 +57,20 @@ protected function doSend(MessageInterface $message): SentMessage
             throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
         }
 
-        $endpoint = \sprintf('https://%s', $this->getEndpoint());
-        $options = $message->getOptions()?->toArray() ?? [];
-        $options['to'] = $message->getRecipientId();
+        $endpoint = sprintf('https://%s', $this->getEndpoint());
 
-        if (!$options['to']) {
-            throw new InvalidArgumentException(\sprintf('The "%s" transport required the "to" option to be set.', __CLASS__));
+        // Generate Options
+        $options = $message->getOptions()?->toArray() ?? [];
+        if (!$options['token'] && !$options['topic']) {
```

Comment:
Is it not `if (!isset($options['token']) && !isset($options['topic'])) {`?

It generates this warning:
```
Warning: Undefined array key “token”
```